### ISSUE TITLE
docker-compose command not showing up in CLI

### ISSUE THREAD

**Issue by Wiedzmin89** [docker/compose#8698]
I installed docker-compose v2.0.0-1
This led to my CLI failed to find "docker-compose" command
It happened because bin file moved from /usr/bin/docker-compose to /usr/lib/docker/cli-plugins/docker-compose

**Comment by Altycoder** [comment]
This seems to have caught a few people out, a bug was raised for Arch today because of this but also because the docker package hasn't been upgraded, so neither docker-compose or docker compose commands work and there was no announcement.

Also mentioned in this previous issue: https://github.com/docker/compose/issues/8697

**Comment by ndeloof** [comment]
> I installed docker-compose v2.0.0-1

How did you? 
Compose V2 is a CLI plugin, installation process is not the same as Compose V1

**Comment by Altycoder** [comment]
> > I installed docker-compose v2.0.0-1
> 
> How did you? Compose V2 is a CLI plugin, installation process is not the same as Compose V1

Well on Arch at least anyone with the docker-compose package was upgraded from v1 to v2 directly so I would so that for Arch that the installation is exactly the same.

**Comment by ndeloof** [comment]
hum, so Arch package system doesn't consider a major version bump is expected to introduce compatibility breaks and require some explicit check?

**Comment by Altycoder** [comment]
> hum, so Arch package system doesn't consider a major version bump is expected to introduce compatibility breaks and require some explicit check?

It seems not and usually when there's a breaking change it's announced on the mailing list so people are forewarned.

**Comment by ndeloof** [comment]
Looking at `https://archlinux.org/packages/community/any/docker-compose/` I can see this has been well set to install compose 2.0.0 as a CLI plugin, but didn't also installed the compose-switch to offer full backward compatibility with `docker-compose` command.

I commented on https://github.com/archlinux/svntogit-community/commit/f9f8d516ae080f1e79a0fa55d6074824c5b4ed75 about this, not sure how to get maintainer attention on this

**Comment by Altycoder** [comment]
ok thanks I'll post it on the reddit thread as well

**Comment by grazzolini** [comment]
To be quite honest, I don't like the idea of including compose-compat/switch into the docker-compose package. I think it should be it's own package.

**Comment by Altycoder** [comment]
Well if it helps I've re-upgraded to v2.0 and the docker compose command does work (yay!) but this will still be a breaking change for anyone with scripts that call docker-compose so a docker-compose-switch package that's a dependency of the docker-compose package would seem like a way foward?

**Comment by ndeloof** [comment]
> It's also absent in the README of this project

Good point. We've been debating a lot on the right way to document using Compose V2 both for newcomers and legacy users who look for backward compatibility. I'll try to better document this.

I'm not familiar with Arch packaging system, can a `compose-switch` package be declared as "recommended" on `docker-compose`? Or can it distinguish major upgrades from 1.* to 2.* vs plain new installation or minor upgrades?

**Comment by ndeloof** [comment]
added a note on release changelog and README: https://github.com/docker/compose#backward-compatibility

**Comment by Foxboron** [comment]
>Good point. We've been debating a lot on the right way to document using Compose V2 both for newcomers and legacy users who look for backward compatibility. I'll try to better document this.

Thanks :) As a packager I usually read the change log on major versions.

>I'm not familiar with Arch packaging system, can a compose-switch package be declared as "recommended" on docker-compose? Or can it distinguish major upgrades from 1.* to 2.* vs plain new installation or minor upgrades?

Almost, we can list "optional dependencies" which would be shown when they update `docker-compose` as a "New optional dependency". It can be described as `docker-compose-switch: v1 backwards compatibility layer` or something.

>added a note on release changelog and README: https://github.com/docker/compose#backward-compatibility

Much better! Thank you :)

**Comment by grazzolini** [comment]
I have several scripts calling docker-compose. So, as a remedy for now I have created a docker-compose script:

`#!/bin/bash
/usr/bin/docker compose $@`

This has been working so far, but I know that some of the V2 commands will be different. It's not meant as a replacement for compose-switch.

**Comment by ndeloof** [comment]
@grazzolini why don't you just download the compose-switch as a real translator vs this tiny script?

**Comment by grazzolini** [comment]
> @grazzolini why don't you just download the compose-switch as a real translator vs this tiny script?

I'm going to probably use the package @Foxboron will work on. For now, this will serve.

**Comment by ndeloof** [comment]
sure, but if you have to create your own `/usr/local/bin/docker-compose` bash script, then just donwload and do the same with compose-switch, being designed for this purpose.

**Comment by grazzolini** [comment]
> sure, but if you have to create your own `/usr/local/bin/docker-compose` bash script, then just donwload and do the same with compose-switch, being designed for this purpose.

Yes, just did so. :+1:

**Comment by Foxboron** [comment]
@ndeloof On the Arch side of things this is now fixed. Plugin has been symlinked to the docker plugin and stuff seems to work. It's been pushed as version `2.0.1-1`.

**Comment by kriansa** [comment]
It is worth noticing that this change not only addresses backward compatibility with those using `docker-compose` directly for `Docker`, but it also solves for those who depends on that binary as the entrypoint for using Compose with `Podman`, which doesn't provide Docker CLI plugin support.

Thanks for working on this so quickly!

**Comment by kostadinnm** [comment]
I can confirm that I'm missing shell completion as well. Running Docker Compose version 2.3.3 on Manjaro latest.

**Comment by ndeloof** [comment]
Missing shell completion is tracked by https://github.com/docker/compose/issues/8550
tl;dr: this require some architectural changes in docker/cli, see https://github.com/docker/cli/pull/3429
